### PR TITLE
support 1.4

### PR DESCRIPTION
**Pull request by paulZzhang**
#529

### CODE REVIEW

**Comment by yordis** [comment]
Hey @paulZzhang thanks a lot for the PR but this is the `rc` version, not an official version of the package.

File: mix.exs [hunk 1]
```diff
@@ -159,7 +159,7 @@ defmodule Guardian.Mixfile do
       {:poison, "~> 2.2 or ~> 3.0"},
 
       # Optional dependencies
-      {:phoenix, "~> 1.0 or ~> 1.2 or ~> 1.3", optional: true},
+      {:phoenix, "~> 1.0 or ~> 1.2 or ~> 1.3 or ~> 1.4.0-rc.3", optional: true},
```
Comment (k-cross) [inline mix.exs]: I've never done this, but is it possible to give maximum flexibility by just changing `... or ~> 1.3` to `... or >= 1.3`?

Comment (yordis) [reply mix.exs]: @k-cross yes it is possible to use `>= 1.3`.

**Comment by paulZzhang** [comment]
@yordis Please publish a rc version.Thanks.

**Comment by yordis** [comment]
@paulZzhang test is failing.

Could you change to be just `>= 1.3`?

File: mix.exs (hunk 1 above)
Comment (k-cross) [reply mix.exs]: Might as well make it `>= 1.0` unless you're expecting to make breaking changes.

**Comment by paulZzhang** [comment]
@yordis  the branch c692ff4 is >=1.3

**Comment by taiansu** [comment]
> @yordis the branch [c692ff4](https://github.com/ueberauth/guardian/commit/c692ff44d0cc08d3c0617d7f76d3e04cf6a0f652) is >=1.3

According to my understanding and experiments, `>= 1.3` is a invalid syntax. We have to specify the patch version number, e.g. `>= 1.3.0` while using the `>=` operator.

In short, `{:phoenix, "~> 1.0 or >= 1.2", optional: true},` works on my machine (elixir 1.7.4, phoenix 1.4.0, erlang 21.1.1).

**Comment by yordis** [comment]
@paulZzhang sorry maybe I wasn't clear about it, I meant to change to ***just*** `>=1.3`.

```
 {:phoenix, ">= 1.3", optional: true},
```

But, @ueberauth/core are talking about this and we will move this dependency outside of `guardian` into `guardian_phoenix` package.

**Comment by doomspork** [comment]
@yordis are you satisfied with this change now?  If so let's merge

**Comment by yordis** [comment]
@doomspork that is Phoenix `1.3` that does not include `1.4` ...

**Comment by jrissler** [comment]
It would be nice to have this merged - I'm in the middle of upgrading to 1.4 now and this is a blocker :)

**Comment by jrissler** [comment]
Thanks!

**Pull request merged**